src/utils/session.ts

```diff
@@ -22,3 +22,26 @@ export function getPresenterAccountDetails(req: Request): Details {
 export function setPresenterAccountDetails(req: Request, details: Details) {
     req.session?.setExtraData(PRESENTER_ACCOUNT_SESSION_KEY, details);
 }
+
+export function generateUserDetails(req: Request): Details{
+    const signinInfo = req.session?.data?.signin_info;
+    const details = {} as Details;
+
+    details.email = signinInfo?.user_profile?.email as string;
+    details.userId = signinInfo?.user_profile?.id as string;
+
+    const forename = signinInfo?.user_profile?.forename as string;
+    const surname = signinInfo?.user_profile?.surname as string;
+
+    const name: Name = { forename, surname };
+
+    details.name = name;
+
+    details.address = req.body;
```

Comment:
The request body needs to have it's type checked to esnure it is correct.
This can be done with the isAddress function in the private sdk.
Ideally, instead of populating the address from the request body, it should be passed into the function as an argument and it should have been checked by the caller.